cpp/tests/iterator/iterator_test.cu

```diff
@@ -144,7 +161,7 @@ TYPED_TEST_CASE(IteratorTest, TestingTypes);
 // tests for non-null iterator (pointer of device array)
 TYPED_TEST(IteratorTest, non_null_iterator)
 {
-    using T = int32_t;
+    using T = TypeParam;
```

Comment:
Currently these tests are only using `int32_t`. This should be updated to use a more complex list of types, perhaps `cudf::test::AllTypes`. The iterator should work on strings, yes?

Reply:
yes. These are legacy tests copied and modified. tests involve sum & square.  some types does not support operator+.  so, I didn't add for other datatypes. iterator should work on string and timestamp_*. I don't know what operations to perform on them. That's why I didn't add them.

Reply:
Can you add new tests that verify the null replacement iterator works as expected for all types?

Reply:
Yes. I will add for `min` or `max` tests for all types.
By the way, `cudf::test::AllTypes` does not include strings now. https://github.com/rapidsai/cudf/blob/edecb21f1cfe8f92f85b6793c61a9961bbc8808c/cpp/tests/utilities/type_lists.hpp#L117

Reply:
> I will add for min or max tests for all types.

I was thinking something simpler than that and independent of any reduction. Something more like you create a null replacement iterator over a nullable column and then you use that iterator to iterate over the column and ensure every value is what it should be (i.e., either the actual value or the replacement value). 

>By the way, cudf::test::AllTypes does not include strings now.
You're right! Could you add `string_view`?

Reply:
`string_view` support for iterator added. 
I didn't add `string_view` to `AllTypes` since it affects other unrelated tests. Better to add as separate PR.

Reply:
all tests are value equal tests.